src/group_theory/finiteness.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2021 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+
+import data.set.finite
+import group_theory.submonoid.operations
+
+/-!
+# Finitely generated monoid.
+
+We define finitely generated monoids.
+
+## Main definition
+
+* `monoid.fg`, `add_monoid.fg` : A monoid is finitely generated.
+
+-/
+
+section monoid
+
+variables {M N : Type*} [monoid M] [add_monoid N]
+
+/-- A submonoid of `M` is finitely generated if it is the closure of a finite subset of `M`. -/
+@[to_additive]
+def submonoid.fg (P : submonoid M) : Prop := ∃ S : finset M, submonoid.closure ↑S = P
+
+/-- An additive submonoid of `N` is finitely generated if it is the closure of a finite subset of
+`M`. -/
+add_decl_doc add_submonoid.fg
+
+variables (M N)
+
+/-- A monoid is finitely generated if it is finitely generated as a submonoid of itself. -/
+@[to_additive]
+def monoid.fg : Prop := (⊤ : submonoid M).fg
+
+/-- An additive monoid is finitely generated if it is finitely generated as an additive submonoid of
+itself. -/
```

Comment:
Here and above, can't you put the docstring as an argument to `to_additive`?

Reply:
Yes we could, but I've seen @urkud do it this way recently because it makes the line wrapping slightly less frustrating, and vs-code does a better job of syntax highlighting it. Having said that I don't feel strongly.